internal/thelma/clients/vault/masking_round_tripper_test.go

```diff
@@ -0,0 +1,31 @@
+package vault
+
+import (
+	vaulthelper "github.com/broadinstitute/thelma/internal/thelma/clients/vault/testing"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func Test_MaskingRoundTripper(t *testing.T) {
+	fakeVaultServer := vaulthelper.NewFakeVaultServer(t)
+	fakeVaultServer.SetSecret("secret/foo/bar", map[string]interface{}{"abc": "xyz"})
+
+	client := fakeVaultServer.Server().Client()
+
+	maskFnCalled := false
+
+	maskingTransport := MaskingRoundTripper{
+		inner: client.Transport,
+		maskFn: func(secrets ...string) {
+			assert.Len(t, secrets, 1)
+			assert.Equal(t, "xyz", secrets[0])
+			maskFnCalled = true
+		},
+	}
+	client.Transport = maskingTransport
+
+	_, err := client.Get(fakeVaultServer.Server().URL + "/v1/secret/foo/bar")
+	require.NoError(t, err)
+	assert.True(t, maskFnCalled)
```

Comment:
this test is a little unclear to me. I think it tests
a) a secret exists
b) it is the expected secret "xyz"
c) maskFn has been called.

I'm not sure what to expect for:
a) is the secret still available to thelma
b) is it just the logging that gets masked.
c) what do i expect to see where there used to be i.e. "xyz": "****"

Reply:
I added some comments to clarify what the test is doing, see these comments for more context: https://github.com/broadinstitute/thelma/pull/39#discussion_r933533209